# PART 11 - MENTAL HEALTH APPS CLEANING - SECOND WEB SCRAPING

In [1]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [2]:
apps_df=pd.read_csv('/Users/nuria/Documents/datasciencecourse/FINAL PROJECT/Apps_mental_health_happiness.csv')

In [3]:
apps_df

App  \
0                                  El Poder de la mente   
1          Audios de Autoestima💪Motivación😄Superación❤️   
2      Goalmap: motivación, bienestar, metas personales   
3                                Crecimiento Espiritual   
4                         Mindfulness Meditación guiada   
...                                                 ...   
1023                           Meditación Guiada Gratis   
1024    Control y Monitor: Ansiedad, Ánimo y Autoestima   
1025            Pixels - Estados de Ánimo y Mindfulness   
1026                          Psicología Positiva Fácil   
1027  PSY online terapia psicologia. Сhat apoyo anón...   

                                  Developer                        Type  \
0                         Emprendedora apps              Estilo de vida   
1                             DAVID B MARÍN                   Educación   
2                                   GOALMAP               Productividad   
3     Apps Bíblicas Cristianas Interesantes  Libros y obras de consulta   
4                                Metta Apps              Estilo de vida   
...                                     ...                         ...   
1023                            AbacaxiApps           Salud y bienestar   
1024                            TwoCats App              Estilo de vida   
1025                              Teo Vogel              Estilo de vida   
1026                      3Temes Publishers           Salud y bienestar   
1027                            INVOLTA INC           Salud y bienestar   

      Rates  Downloads+                                       Last_reviews  
0       4.4    100.000+  Gauuu, que buena aplicación. Amo los temas de ...  
1       4.6      5.000+  Muy buenos mensajes, positivos y con mucho emp...  
2       3.3    100.000+  Excelente aplicación para cumplir metas de háb...  
3       4.6    100.000+  Muy buena aplicasion,lo que más me gusta es qu...  
4       4.1    100.000+  El contenido de la app es excepcional y muy út...  
...     ...         ...                                                ...  
1023    4.3     10.000+  Estan muy interesantes las meditaciones guiada...  
1024    4.6  1.000.000+  Esta es de las mejores cosas que me pasó, es u...  
1025    4.7    500.000+  me gusta mucho la app, pero estaría bueno que ...  
1026    4.1     50.000+  letras muy pequeñas, que se hace imposible lee...  
1027    4.3    500.000+  No me permite abrir la app y me gustaría segui...  

[1028 rows x 6 columns]

##### CHECK FOR NULL VALUES

In [4]:
apps_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1028 entries, 0 to 1027
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   App           1028 non-null   object 
 1   Developer     1028 non-null   object 
 2   Type          1028 non-null   object 
 3   Rates         1028 non-null   float64
 4   Downloads+    1028 non-null   object 
 5   Last_reviews  1026 non-null   object 
dtypes: float64(1), object(5)
memory usage: 48.3+ KB


We can see that there are two missing reviews
- Mental health and happiness
- Motivational
- Self esteem
- Personal grouth

##### CHECK DUPLICATES

In [7]:
apps_df.duplicated().sum() 

336

There are 336 duplicated apps, that is possible because when I was web scrapping Google Play for the diffetent possible entries in the searcher some apps appeared in the different searches

In [8]:
apps_df.drop_duplicates(inplace=True)

In [9]:
apps_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 692 entries, 0 to 1027
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   App           692 non-null    object 
 1   Developer     692 non-null    object 
 2   Type          692 non-null    object 
 3   Rates         692 non-null    float64
 4   Downloads+    692 non-null    object 
 5   Last_reviews  690 non-null    object 
dtypes: float64(1), object(5)
memory usage: 37.8+ KB


In [13]:
apps_df.App.nunique()

156

I'm going to study 156 different Apps to imporive happiness and personal wellness

##### CONVERT THE DOWNLOAD COLUMN INTO NUMERICAL VARIABLE

In [17]:
apps_df.reset_index(drop=True,inplace=True)

In [19]:
apps_df.head()

App  \
0                              El Poder de la mente   
1      Audios de Autoestima💪Motivación😄Superación❤️   
2  Goalmap: motivación, bienestar, metas personales   
3                            Crecimiento Espiritual   
4                     Mindfulness Meditación guiada   

                               Developer                        Type  Rates  \
0                      Emprendedora apps              Estilo de vida    4.4   
1                          DAVID B MARÍN                   Educación    4.6   
2                                GOALMAP               Productividad    3.3   
3  Apps Bíblicas Cristianas Interesantes  Libros y obras de consulta    4.6   
4                             Metta Apps              Estilo de vida    4.1   

  Downloads+                                       Last_reviews  
0   100.000+  Gauuu, que buena aplicación. Amo los temas de ...  
1     5.000+  Muy buenos mensajes, positivos y con mucho emp...  
2   100.000+  Excelente aplicación para cumplir metas de háb...  
3   100.000+  Muy buena aplicasion,lo que más me gusta es qu...  
4   100.000+  El contenido de la app es excepcional y muy út...

In [21]:
apps_df['Downloads+'].unique()

array(['100.000+', '5.000+', '10.000+', '1.000+', '1.000.000+', '11M',
       '500.000+', '50.000+', '5.000.000+', '10.000.000+', '100+', '500+'],
      dtype=object)

In [22]:
apps_df[apps_df['Downloads+']=='11M']

App   Developer            Type  Rates Downloads+  \
18   Coaching Personal PRO  Metta Apps  Estilo de vida    4.9        11M   
51   Coaching Personal PRO  Metta Apps  Estilo de vida    4.9        11M   
105  Coaching Personal PRO  Metta Apps  Estilo de vida    4.9        11M   
167  Coaching Personal PRO  Metta Apps  Estilo de vida    4.9        11M   

                                          Last_reviews  
18                                               Buena  
51   He encontrado una gran fuente de inspiración. ...  
105  Excelente lo que estaba buscando. Felicitacion...  
167  Plantearte preguntas y auto responderse es un ...

In [26]:
apps_df.iloc[18,4]='11.000'
apps_df.iloc[51,4]='11.000'
apps_df.iloc[105,4]='11.000'
apps_df.iloc[167,4]='11.000'

In [28]:
apps_df['Downloads+'] = apps_df.loc[:,'Downloads+'].str.replace('+', '')

In [29]:
apps_df.head()

App  \
0                              El Poder de la mente   
1      Audios de Autoestima💪Motivación😄Superación❤️   
2  Goalmap: motivación, bienestar, metas personales   
3                            Crecimiento Espiritual   
4                     Mindfulness Meditación guiada   

                               Developer                        Type  Rates  \
0                      Emprendedora apps              Estilo de vida    4.4   
1                          DAVID B MARÍN                   Educación    4.6   
2                                GOALMAP               Productividad    3.3   
3  Apps Bíblicas Cristianas Interesantes  Libros y obras de consulta    4.6   
4                             Metta Apps              Estilo de vida    4.1   

  Downloads+                                       Last_reviews  
0    100.000  Gauuu, que buena aplicación. Amo los temas de ...  
1      5.000  Muy buenos mensajes, positivos y con mucho emp...  
2    100.000  Excelente aplicación para cumplir metas de háb...  
3    100.000  Muy buena aplicasion,lo que más me gusta es qu...  
4    100.000  El contenido de la app es excepcional y muy út...

In [30]:
apps_df['Downloads+'] = apps_df.loc[:,'Downloads+'].str.replace('.', '')

In [31]:
apps_df.head()

App  \
0                              El Poder de la mente   
1      Audios de Autoestima💪Motivación😄Superación❤️   
2  Goalmap: motivación, bienestar, metas personales   
3                            Crecimiento Espiritual   
4                     Mindfulness Meditación guiada   

                               Developer                        Type  Rates  \
0                      Emprendedora apps              Estilo de vida    4.4   
1                          DAVID B MARÍN                   Educación    4.6   
2                                GOALMAP               Productividad    3.3   
3  Apps Bíblicas Cristianas Interesantes  Libros y obras de consulta    4.6   
4                             Metta Apps              Estilo de vida    4.1   

  Downloads+                                       Last_reviews  
0     100000  Gauuu, que buena aplicación. Amo los temas de ...  
1       5000  Muy buenos mensajes, positivos y con mucho emp...  
2     100000  Excelente aplicación para cumplir metas de háb...  
3     100000  Muy buena aplicasion,lo que más me gusta es qu...  
4     100000  El contenido de la app es excepcional y muy út...

In [32]:
apps_df['Downloads+']=apps_df['Downloads+'].astype(int)

In [33]:
apps_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 692 entries, 0 to 691
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   App           692 non-null    object 
 1   Developer     692 non-null    object 
 2   Type          692 non-null    object 
 3   Rates         692 non-null    float64
 4   Downloads+    692 non-null    int64  
 5   Last_reviews  690 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 32.6+ KB


In [34]:
apps_df.to_csv('Apps_happiness_mental_health_cleaned_def.csv',index=False)